include/deal.II/dofs/dof_objects.h

```diff
@@ -203,7 +204,7 @@ namespace internal
     {
       (void)fe_index;
       Assert(
-        (fe_index == dealii::DoFHandler<dh_dim, spacedim>::default_fe_index),
+        (fe_index == 0),
```

Comment:
I guess you can change this line back to use the default_fe_index since you turned both default and invalid dof indices to unique values.
```suggestion
        (fe_index == dealii::DoFHandler<dh_dim, spacedim>::default_fe_index),
```